homeassistant/components/netatmo/data_handler.py

```diff
@@ -143,42 +136,53 @@ async def async_fetch_data(self, data_class, data_class_entry, **kwargs):
         except pyatmo.ApiError as err:
             _LOGGER.debug(err)
 
+        except asyncio.exceptions.TimeoutError as err:
+            _LOGGER.debug(err)
+            return
+
+        for update_callback in self.data_classes[data_class_entry]["subscriptions"]:
+            if update_callback:
```

Comment:
Why do we need this check?

Reply:
When setting up the platform I register the data class with `None` for the callback and use this in the `netatmo_entity_base.py` to clean up once the actual entities are set up. This is done to ensure there are no unused data classes floating around and creating unnecessary API calls. I haven't revisited that code to see if there is a more elegant solution.